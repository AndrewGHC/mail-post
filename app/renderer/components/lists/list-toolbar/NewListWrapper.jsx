import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui/styles';
import Paper from 'material-ui/Paper';

import NewList from './NewList';

const styles = theme => ({
  root: theme.mixins.gutters({
    paddingTop: 16,
    paddingBottom: 16,
    marginBottom: theme.spacing.unit * 3,
  }),
});

const NewListWrapper = ({
  classes,
  ...rest
}) => (
  <Paper className={classes.root}>
    <NewList {...rest} />
  </Paper>
);

NewListWrapper.propTypes = {
  classes: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
};

export default withStyles(styles)(NewListWrapper);
