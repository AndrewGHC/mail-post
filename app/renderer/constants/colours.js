import blueGrey from '@material-ui/core/colors/blueGrey';
import teal from '@material-ui/core/colors/teal';
import grey from '@material-ui/core/colors/grey';

export const COLOUR_PRIMARY = teal[600];
export const COLOUR_SECONDARY = blueGrey[900];
export const COLOUR_BLACK = grey[900];
export const COLOUR_WHITE = grey[100];
