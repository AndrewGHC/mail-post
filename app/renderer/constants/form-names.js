export const FORM_NEW_LIST = 'FORM_NEW_LIST'; // eslint-disable-line import/prefer-default-export
