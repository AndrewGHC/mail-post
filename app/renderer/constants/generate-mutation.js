/* eslint-disable import/prefer-default-export */
import { QUERY_GET_LISTS } from './queries';

export const generateImportCsv = (csvPath, name) => ({
  variables: {
    csvPath,
    name,
  },
  update: (store, { data: { importCsv } }) => {
    const data = store.readQuery({ query: QUERY_GET_LISTS });
    data.lists.push(importCsv);
    store.writeQuery({ query: QUERY_GET_LISTS, data });
  },
});
