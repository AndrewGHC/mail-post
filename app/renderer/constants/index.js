export * from './routes';
export * from './colours';
export * from './form-names';
