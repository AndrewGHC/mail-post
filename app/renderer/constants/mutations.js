/* eslint-disable import/prefer-default-export */
import gql from 'graphql-tag';

export const MutationCreateListCsv = gql`
  mutation MutationCreateListCsv($csvPath: String!, $name: String!) {
    importCsv(csvPath: $csvPath, name: $name) {
      id
      name
      total_subscribers
    }
  }
`;
