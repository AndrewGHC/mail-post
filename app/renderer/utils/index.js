export { default as injectGlobalStyles } from './inject-global-styles';
export { default as io } from './io';
