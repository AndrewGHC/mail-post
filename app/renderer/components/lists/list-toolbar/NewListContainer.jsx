import React from 'react';
import PropTypes from 'prop-types';
import { reduxForm, formValueSelector } from 'redux-form';
import { connect } from 'react-redux';
import { compose } from 'ramda';

import { FORM_NEW_LIST } from '../../../constants';

import NewList from './NewList';

const nameOfList = 'newList';

const NewListContainer = ({ listNameValue, nameOfListProp, ...rest }) => (
  <NewList
    listNameValue={listNameValue}
    nameOfList={nameOfListProp}
    {...rest}
  />
);

NewListContainer.propTypes = {
  listNameValue: PropTypes.string.isRequired,
  nameOfListProp: PropTypes.string,
};

NewListContainer.defaultProps = {
  nameOfListProp: nameOfList,
};

const selectForm = formValueSelector(FORM_NEW_LIST);

const mapStateToProps = state => ({
  listNameValue: selectForm(state, nameOfList) || '',
});

export default compose(
  connect(mapStateToProps),
  reduxForm({
    form: FORM_NEW_LIST,
    destroyOnUnmount: false,
  }),
)(NewListContainer);
