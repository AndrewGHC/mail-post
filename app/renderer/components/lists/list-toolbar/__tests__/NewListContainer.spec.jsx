import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { ApolloProvider } from 'react-apollo';
import configureStore from 'redux-mock-store';

import { FORM_NEW_LIST } from '../../../../constants';
import NewListContainer from '../NewListContainer';
import { apolloClient } from '../../../../utils';

import NewListWrapper from '../NewListWrapper';

const mockStore = configureStore();
const listNameValue = 'hello';
describe('NewListContainer', () => {
  let wrapper;
  let store;
  let state;

  beforeEach(() => {
    state = {
      form: {
        [FORM_NEW_LIST]: {
          values: {
            newList: 'hello',
          },
        },
      },
    };
    store = mockStore(state);
    wrapper = mount(
      <ApolloProvider client={apolloClient}>
        <Provider store={store}>
          <NewListContainer />
        </Provider>
      </ApolloProvider>,
    );
  });

  it('passes listNameValue and nameOfList from store to NewListWrapper', () => {
    const props = {
      listNameValue,
      nameOfList: 'newList',
    };

    expect(wrapper.find(NewListWrapper).props()).toEqual(expect.objectContaining(props));
  });
});
