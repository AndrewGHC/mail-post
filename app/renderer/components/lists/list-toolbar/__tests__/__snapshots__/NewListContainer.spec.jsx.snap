// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewListContainer matches snapshot 1`] = `
<Connect(Form(NewListContainer))
  destroyOnUnmount={false}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="FORM_NEW_LIST"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  listNameValue="hello"
  newListName="hi"
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
/>
`;
